### PR TITLE
Contact request bugs (#433)

### LINKED ISSUE #433: Contact request
1. Don't add contact to the contacts list until `contact request` will be approved
2. We need some way to "remind" user about pending `contact requests` (in chat, chats screens or elsewhere)

### PR DESCRIPTION
**Pull request by rasom**
fix #433

### CODE REVIEW

**Comment by rasom** [comment]
The open question is how to remind the user that he hasn't added the contact. Unless this PR is ready

**Comment by jarradh** [comment]
@andmironov do you have any ideas, like maybe a small pop-up notification/tab that points to chat options? or some tab that does add to contacts or ignore?

**Comment by andmironov** [comment]
Here's one way of showing the reminder
add_to_contacts

**Comment by andmironov** [comment]
Here's another one (Zeplin https://zpl.io/Z2mwyPP )

add_to_contacts2

**Comment by rasom** [comment]
What to do if we have pending dapp contact? Show that small "pending" badge? I accidentally added both dapp and pending badges to `Console`, it looks weird :)

screen shot 2016-11-15 at 12 24 35

**Comment by jarradh** [comment]
oh dear :)

**Comment by rasom** [comment]
That's how it looks now

screen shot 2016-11-15 at 13 05 26

screen shot 2016-11-15 at 13 05 36

**Comment by alwx** [comment]
What do I need to do to see this item in the message list?
screen shot 2016-11-15 at 17 12 35

**Comment by alwx** [comment]
I think you should also remove "Add to contacts" from the contact menu, because it is redundant.

screen shot 2016-11-15 at 17 14 08

File: src/status_im/chat/screen.cljs [hunk 1]
```diff
@@ -106,8 +120,8 @@
    [toolbar {:hide-nav?      (or (empty? accounts) show-actions?)
              :custom-content [toolbar-content-view]
              :custom-action  [toolbar-action]
-             :style          (get-in platform-specific [:component-styles :toolbar])}]])
-
+             :style          (get-in platform-specific [:component-styles :toolbar])}]
+   [add-conact-bar]])
```
Comment (alwx) [inline src/status_im/chat/screen.cljs]: typo

File: src/status_im/chat/screen.cljs [hunk 2]
```diff
@@ -185,25 +199,25 @@
...
-          (when-not @command?
-            [suggestion-container])
-          [response-view]
-          [chat-message-input-view]
-          (when @show-actions?
-            [actions-view])
-          (when @show-bottom-info?
-            [bottom-info-view])
-          [offline-view {:top (get-in platform-specific [:component-styles :status-bar :default :height])}]])})))
+                            (fn []
```
Comment (alwx) [inline src/status_im/chat/screen.cljs]: Is there a tabulation problem here?

**Comment by rasom** [comment]
There was two option how to show contact request, I implemented the first.

File: src/status_im/data_store/realm/schemas/account/v4/core.cljs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+             kv-store/schema
+             message/schema
+             pending-message/schema
+             request/schema
+             tag/schema
+             user-status/schema])
+
+(defn migration [old-realm new-realm]
+  (log/debug "migrating v4 account database: " old-realm new-realm)
+  (contact/migration old-realm new-realm)
```
Comment (alwx) [inline src/status_im/data_store/realm/schemas/account/v4/core.cljs]: there is nothing inside this contact migration

Comment (rasom) [inline src/status_im/data_store/realm/schemas/account/v4/core.cljs]: https://github.com/status-im/status-react/pull/449/files#diff-04e9cde5935844be1b381285f962bb85R35

Comment (alwx) [inline src/status_im/data_store/realm/schemas/account/v4/core.cljs]: what you're showing is chat migration and it is fine
but there is no need in contact migration

**Comment by jarradh** [comment]
How does the add to contacts tab work when offline is also shown?

**Comment by rasom** [comment]
that's how it looks with offline tab

screen shot 2016-11-16 at 14 13 25

**Comment by rasom** [comment]
Is this version better?

screen shot 2016-11-16 at 14 13 25

**Comment by rasom** [comment]
Is this version better?

screen shot 2016-11-16 at 14 25 04

**Comment by jarradh** [comment]
it's good!

**Comment by rasom** [comment]
@alwx ready for review

**Comment by alwx** [comment]
@rasom Looks okay now. You can merge it.

**Pull request merged**